src/context/MessageBounceContext.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { createContext, ReactEventHandler, useCallback, useContext, useMemo } from 'react';
+import { useMessageContext } from './MessageContext';
+import { DefaultStreamChatGenerics, PropsWithChildrenOnly } from '../types/types';
+import { StreamMessage } from './ChannelStateContext';
+import { useChannelActionContext } from './ChannelActionContext';
+import { isMessageBounced } from '../components';
+
+export interface MessageBounceContextValue<
+  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics
+> {
+  handleDelete: ReactEventHandler;
+  handleEdit: ReactEventHandler;
+  handleRetry: ReactEventHandler;
+  message: StreamMessage<StreamChatGenerics>;
+}
+
+const MessageBounceContext = createContext<MessageBounceContextValue | undefined>(undefined);
+
+export function useMessageBounceContext<
+  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics
+>(componentName?: string) {
+  const contextValue = useContext(MessageBounceContext);
+
+  if (!contextValue) {
+    console.warn(
+      `The useMessageInputContext hook was called outside of the MessageInputContext provider. Make sure this hook is called within the MessageInput's UI component. The errored call is located in the ${componentName} component.`,
+    );
```

Comment:
```suggestion
      `The useMessageBounceContext hook was called outside of the MessageBounceContext provider. The errored call is located in the ${componentName} component.`,
    );
```

Reply:
Blind copy-paste strikes again, thanks for the catch!